contracts/core/EntryPoint.sol

```diff
@@ -818,4 +816,14 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
         (bool success, bytes memory ret) = target.delegatecall(data);
         revert DelegateAndRevert(success, ret);
     }
+
+    function _getUnusedGasPenalty(uint256 gasUsed, uint256 gasLimit) internal pure returns (uint256) {
+        if (gasLimit > gasUsed) {
```

Comment:
How can it be the other way around?

Also, flip this `if` to exit early with 0 please

Reply:
Fixed